### PR TITLE
Add player id tracking

### PR DESCRIPTION
**Pull request by matte-ek**
Allows the user to do a lot of things, such as getting the player avatar, or use the osu!api for the user. Please note that the id will only get filled in after `!mp settings` has been ran

### CODE REVIEW

File: BanchoSharp/Multiplayer/MultiplayerLobby.cs [hunk 1]
```diff
@@ -417,12 +417,19 @@ private void UpdatePlayerInformation(string banchoResponse)
 		// Bancho may send extra player info after the name, for example "[Host / HardRock]", after the 16
 		// character player name bit.
 		string? playerInfo = banchoResponse.Length > (playerNameBegin + 16) ? banchoResponse[(playerNameBegin + 16)..] : null;
-
+		
+		// Find the digits from "/u/" to where the name begins
+		int playerId = int.Parse(banchoResponse[banchoResponse.IndexOf("/u/", StringComparison.Ordinal)..(playerNameBegin-1)].Where(char.IsDigit).ToArray());
+
```
Comment (hburn7) [inline BanchoSharp/Multiplayer/MultiplayerLobby.cs]: Put this in a `.TryParse` instead, if for some reason it doesn't work we don't want to crash.

File: BanchoSharp/Multiplayer/MultiplayerLobby.cs [hunk 2]
```diff
@@ -417,12 +417,19 @@ private void UpdatePlayerInformation(string banchoResponse)
...
+		int playerId = int.Parse(banchoResponse[banchoResponse.IndexOf("/u/", StringComparison.Ordinal)..(playerNameBegin-1)].Where(char.IsDigit).ToArray());
+		
 		var player = FindPlayer(playerName);
 
 		if (player is null)
 		{
-			player = new MultiplayerPlayer(playerName, slot);
+			player = new MultiplayerPlayer(playerName, slot)
+			{
+				Id = playerId
```
Comment (hburn7) [inline BanchoSharp/Multiplayer/MultiplayerLobby.cs:435]: We can save a few lines: no need to specify the Id on init if we are force setting it anyway.

File: BanchoSharp/Multiplayer/MultiplayerPlayer.cs [hunk 3]
```diff
@@ -9,6 +9,8 @@ public enum TeamColor
 
 public class MultiplayerPlayer
 {
+	public int? Id { get; set; }
+
```
Comment (hburn7) [inline BanchoSharp/Multiplayer/MultiplayerPlayer.cs]: Line should be removed

**Review by hburn7 — CHANGES_REQUESTED** [review]

File: BanchoSharp/Multiplayer/MultiplayerPlayer.cs (hunk 3 above)
Comment (hburn7) [reply BanchoSharp/Multiplayer/MultiplayerPlayer.cs]: Line 13, to be specific.

File: BanchoSharp/Multiplayer/MultiplayerLobby.cs (hunk 2 above)
Comment (matte-ek) [reply BanchoSharp/Multiplayer/MultiplayerLobby.cs:435]: Then when `OnPlayerJoined` gets invoked, the player doesn't have a valid id, even though we already have the data.

File: BanchoSharp/Multiplayer/MultiplayerLobby.cs (hunk 1 above)
Comment (matte-ek) [reply BanchoSharp/Multiplayer/MultiplayerLobby.cs]: I mean sure, I wasn't considering it since we do `.Parse` at multiple other places in the parsing code, and this should **never** be invalid in the first place, I'll be ready to eat my own words if that were to happen.

Comment (hburn7) [reply BanchoSharp/Multiplayer/MultiplayerLobby.cs]: Yeah, we should work on turning any `.Parse` methods into `.TryParse` for the same reason. We can't guarantee that BanchoBot's messages will remain the same, or that there isn't an edge case we haven't seen before, etc. You just never know. If we were creating the input ourselves, we could use `.Parse` and validate with a test case, but because it's coming from an outside source out of our control, it's not guaranteed.

**Review by hburn7 — APPROVED** [review]
